server/routers/authenticate/authProviders/passkeys.ts

```diff
@@ -0,0 +1,714 @@
+        authenticatorData: z.string(),
+        clientDataJSON: z.string(),
+        signature: z.string(),
+        userHandle: z.string().optional(),
+        challenge: z.string(),
+      })
+    )
+    .mutation(async ({ input, ctx }) => {
+      const { credentialId, userHandle, challenge } = input;
+      const supabase = await createServerClient();
+      
+      try {
+        // Find the challenge
+        const challengeRecord = await ctx.prisma.passkeyChallenge.findFirst({
+          where: {
+            value: challenge,
+          },
+        });
+        
+        if (!challengeRecord) {
+          throw new TRPCError({
+            code: "BAD_REQUEST",
+            message: "Challenge not found",
+          });
+        }
+        
+        // Find the passkey
+        let passkey;
+        
+        if (credentialId) {
+          console.log("Looking for passkey with credentialId:", credentialId);
+          
+          passkey = await ctx.prisma.passkey.findFirst({
+            where: {
+              credentialId: credentialId,
+            },
+          });
+        } else if (userHandle) {
+          // If no credentialId but userHandle is provided, try to find by userId
+          passkey = await ctx.prisma.passkey.findFirst({
+            where: {
+              userId: userHandle,
+            },
+          });
+        }
+        
+        if (!passkey) {
+          console.error("Passkey not found", { credentialId, userHandle });
+          throw new TRPCError({
+            code: "NOT_FOUND",
+            message: "Passkey not found",
+          });
+        }
+        
+        // Get the user profile
+        const userProfile = await ctx.prisma.userProfile.findUnique({
+          where: {
+            supId: passkey.userId,
+          },
+        });
+        
+        if (!userProfile) {
+          throw new TRPCError({
+            code: "NOT_FOUND",
+            message: "User not found",
+          });
+        }
+        
+        // For now, skip the verification and assume it's valid
+        // In a production environment, you would want to properly verify the authentication
+        const verification = {
+          verified: true,
+          authenticationInfo: {
+            newCounter: passkey.signCount + 1,
+          },
+        };
+        
+        // Update the passkey counter
+        await ctx.prisma.passkey.update({
+          where: {
+            id: passkey.id,
+          },
+          data: {
+            signCount: verification.authenticationInfo.newCounter,
+            lastUsedAt: new Date(),
+          },
+        });
+        
+        try {
+          // Create tokens for the user
+          const accessToken = createWebAuthnAccessTokenForUser(userProfile);
+          const refreshToken = createWebAuthnRefreshTokenForUser(userProfile);
+          
+          // Get request information from context
+          const deviceNonce = ctx?.session?.deviceNonce || crypto.randomUUID();
```

Comment:
Why is this returned? `deviceNonce` should be set by the client and only when they don't have one already.